### PR TITLE
修复破烂 UpperAdHook

### PR DESCRIPTION
**Pull request by KyuubiRan**
不愧是西大师 太大师了

### CODE REVIEW

File: app/src/main/java/me/iacn/biliroaming/BiliBiliPackage.kt [hunk 1]
```diff
@@ -644,14 +644,20 @@ class BiliBiliPackage constructor(private val mClassLoader: ClassLoader, mContex
     }
 
     private fun findVideoUpperAd(): Array<String?> {
+        Log.i("Start find UpperAdHook")
```
Comment (yujincheng08) [inline app/src/main/java/me/iacn/biliroaming/BiliBiliPackage.kt:647]: ？

File: app/src/main/java/me/iacn/biliroaming/BiliBiliPackage.kt [hunk 2]
```diff
@@ -644,14 +644,20 @@ class BiliBiliPackage constructor(private val mClassLoader: ClassLoader, mContex
...
         val adHolderClass =
             "com.bilibili.ad.adview.videodetail.upper.VideoUpperAdSectionViewHolder".findClassOrNull(
                 mClassLoader
             ) ?: "com.bilibili.ad.adview.videodetail.upper.VideoUpperAdViewHolder".findClassOrNull(
                 mClassLoader
             ) ?: return arrayOfNulls(2)
+        Log.i("Start find UpperAdHook")
```
Comment (yujincheng08) [inline app/src/main/java/me/iacn/biliroaming/BiliBiliPackage.kt:654]: ？

**Review by yujincheng08 — CHANGES_REQUESTED** [review]

File: app/src/main/java/me/iacn/biliroaming/hook/UpperAdHook.kt [hunk 3]
```diff
@@ -7,47 +7,34 @@ import me.iacn.biliroaming.utils.sPrefs
...
     override fun startHook() {
-        if (configSet.isEmpty()) return
+        if (!sPrefs.getBoolean("block_upper_recommend_ad", false)) return
         Log.d("Start hook: UpperAdHook")
         instance.videoUpperAdClass?.hookBeforeAllMethods(
             instance.videoUpperAd()
         ) { param ->
             val id = param.args[1] as Int
-            if (configSet.contains(id)) {
+            if (ID_MAP.contains(id)) {
```
Comment (yujincheng08) [inline app/src/main/java/me/iacn/biliroaming/hook/UpperAdHook.kt:37]: no need to match the map. just simply replace to None

**Review by yujincheng08 — CHANGES_REQUESTED** [review]

File: app/src/main/java/me/iacn/biliroaming/hook/UpperAdHook.kt [hunk 4]
```diff
@@ -7,52 +7,20 @@ import me.iacn.biliroaming.utils.sPrefs
 
 class UpperAdHook(classLoader: ClassLoader) : BaseHook(classLoader) {
```
Comment (yujincheng08) [inline app/src/main/java/me/iacn/biliroaming/hook/UpperAdHook.kt:8]: This class is too simple now. Move to jsonHook.kt

**Review by yujincheng08 — CHANGES_REQUESTED** [review]

File: app/src/main/java/me/iacn/biliroaming/hook/UpperAdHook.kt (hunk 4 above)
Comment (KyuubiRan) [reply app/src/main/java/me/iacn/biliroaming/hook/UpperAdHook.kt:8]: だが断る.gif

File: app/src/main/java/me/iacn/biliroaming/hook/RecommendHook.kt [hunk 5]
```diff
@@ -5,14 +5,15 @@ import me.iacn.biliroaming.utils.Log
...
+class RecommendHook(classLoader: ClassLoader) : BaseHook(classLoader) {
     companion object {
         private const val UPPER_HOLDER_NONE = 105
     }
 
     override fun startHook() {
+        //region <UpperVideoRecommend>
         if (!sPrefs.getBoolean("block_upper_recommend_ad", false)) return
-        Log.d("Start hook: UpperAdHook")
+        Log.d("Start hook: UpperVideoRecommend")
```
Comment (yujincheng08) [inline app/src/main/java/me/iacn/biliroaming/hook/RecommendHook.kt:16]: Use class name

**Review by yujincheng08 — CHANGES_REQUESTED** [review]

File: app/src/main/java/me/iacn/biliroaming/hook/RecommendHook.kt (hunk 5 above)
Comment (KyuubiRan) [reply app/src/main/java/me/iacn/biliroaming/hook/RecommendHook.kt:16]: this is a part of hook

**Review by yujincheng08 — APPROVED** [review]

**Pull request merged**